tests/Builder/Expression/RegexFindAllOperatorTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Expression;
+
+use MongoDB\BSON\Regex;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+/**
+ * Test $regexFindAll expression
+ */
+class RegexFindAllOperatorTest extends PipelineTestCase
+{
+    public function testIOption(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::addFields(
+                returnObject: Expression::regexFindAll(
+                    input: Expression::stringFieldPath('description'),
+                    regex: new Regex('line', 'i'),
+                ),
+            ),
+            Stage::addFields(
+                returnObject: Expression::regexFindAll(
+                    input: Expression::stringFieldPath('description'),
+                    regex: 'line',
+                    options: 'i',
+                ),
+            ),
+            Stage::addFields(
+                returnObject: Expression::regexFindAll(
+                    input: Expression::stringFieldPath('description'),
+                    regex: new Regex('line'),
+                    options: 'i',
```

Comment:
Should there be a manual test for the invalid input when the regex BSON type includes options and the `$regexFindAll` `options` option is also specified? Or do you rely on the server to catch that?

Reply:
This would have the advantage of triggering an error early on instead of waiting for the command to be executed.
This check can be in the factory method, as it's the only place we can alter the code before generated code.